--- origsrc/mate-notification-daemon-1.4.0/src/daemon/daemon.c	2012-07-05 13:29:58.000000000 -0500
+++ src/mate-notification-daemon-1.4.0/src/daemon/daemon.c	2012-12-25 22:17:54.238694300 -0600
@@ -99,7 +99,6 @@ typedef struct {
 struct _NotifyDaemonPrivate {
 	guint next_id;
 	guint timeout_source;
-	guint exit_timeout_source;
 	GHashTable* idle_reposition_notify_ids;
 	GHashTable* monitored_window_hash;
 	GHashTable* notification_hash;
@@ -152,35 +151,6 @@ static void _notify_timeout_destroy(Noti
 	g_free(nt);
 }
 
-static gboolean do_exit(gpointer user_data)
-{
-	//g_debug("Exiting due to inactivity");
-	exit(1);
-
-	return FALSE;
-}
-
-static void add_exit_timeout(NotifyDaemon* daemon)
-{
-    if (daemon->priv->exit_timeout_source > 0)
-    {
-		return;
-	}
-
-	daemon->priv->exit_timeout_source = g_timeout_add_seconds(IDLE_SECONDS, do_exit, NULL);
-}
-
-static void remove_exit_timeout(NotifyDaemon* daemon)
-{
-	if (daemon->priv->exit_timeout_source == 0)
-	{
-		return;
-	}
-
-	g_source_remove(daemon->priv->exit_timeout_source);
-	daemon->priv->exit_timeout_source = 0;
-}
-
 static void create_stack_for_monitor(NotifyDaemon* daemon, GdkScreen* screen, int monitor_num)
 {
 	int screen_num = gdk_screen_get_number(screen);
@@ -364,8 +334,6 @@ static void notify_daemon_init(NotifyDae
 	daemon->priv->next_id = 1;
 	daemon->priv->timeout_source = 0;
 
-	add_exit_timeout(daemon);
-
 	client = mateconf_client_get_default();
 	mateconf_client_add_dir(client, MATECONF_KEY_DAEMON, MATECONF_CLIENT_PRELOAD_NONE, NULL);
 
@@ -434,8 +402,6 @@ static void notify_daemon_finalize(GObje
 		gdk_window_remove_filter(NULL, (GdkFilterFunc) _notify_x11_filter, daemon);
 	}
 
-	remove_exit_timeout(daemon);
-
 	g_hash_table_destroy(daemon->priv->monitored_window_hash);
 	g_hash_table_destroy(daemon->priv->idle_reposition_notify_ids);
 	g_hash_table_destroy(daemon->priv->notification_hash);
@@ -533,11 +499,6 @@ static void _close_notification(NotifyDa
 		}
 
 		g_hash_table_remove(priv->notification_hash, &id);
-
-		if (g_hash_table_size(daemon->priv->notification_hash) == 0)
-		{
-			add_exit_timeout(daemon);
-		}
 	}
 }
 
@@ -755,11 +716,6 @@ static gboolean _check_expiration(Notify
 	if (!has_more_timeouts)
 	{
 		daemon->priv->timeout_source = 0;
-
-		if (g_hash_table_size (daemon->priv->notification_hash) == 0)
-		{
-			add_exit_timeout(daemon);
-		}
 	}
 
 	return has_more_timeouts;
@@ -837,7 +793,6 @@ static NotifyTimeout* _store_notificatio
 	_calculate_timeout(daemon, nt, timeout);
 
 	g_hash_table_insert(priv->notification_hash, g_memdup(&id, sizeof(guint)), nt);
-	remove_exit_timeout(daemon);
 
 	return nt;
 }
